qemu/tests/virtio_fs_group_permission_access.py

```diff
@@ -0,0 +1,149 @@
+        cmd_run_virtiofsd += " --shared-dir %s" % shared_dir
+        error_context.context('Running daemon command %s' % cmd_run_virtiofsd,
+                              test.log.info)
+
+        host_session = aexpect.ShellSession("runuser -l " + user_two +
+                                            " -c '" + cmd_run_virtiofsd + "'",
+                                            auto_close=False,
+                                            output_func=utils_misc.log_line,
+                                            output_params=('virtiofs_fs-virtiofs.log',),
+                                            prompt=r"^\[.*\][\#\$]\s*$")
+        params["fs_source_base_dir"] = "/home/" + user_one
+
+        vm = env.get_vm(params.get("main_vm"))
+        vm.create()
+        vm.verify_alive()
+        guest_session = vm.wait_for_login()
+
+        if windows:
+            guest_session = utils_test.qemu.windrv_check_running_verifier(guest_session,
+                                                                          vm,
+                                                                          test,
+                                                                          driver_name)
+            virtio_fs_utils.run_viofs_service(test, params, guest_session)
+            guest_session = vm.reboot(guest_session)
+        else:
+            error_context.context("Create a destination directory %s "
+                                  "inside guest." % fs_dest, test.log.info)
+            if not utils_misc.make_dirs(fs_dest, session=guest_session):
+                test.fail("Creating directory was failed!")
+            error_context.context("Mount virtiofs target %s to %s inside"
+                                  " guest." % (fs_target, fs_dest),
+                                  test.log.info)
+            if not utils_disk.mount(fs_target, fs_dest, 'virtiofs',
+                                    session=guest_session):
+                test.fail('Mount virtiofs target failed.')
+        virtio_fs_utils.basic_io_test(test, params, guest_session)
+    finally:
+        if guest_session:
+            if windows:
+                virtio_fs_utils.delete_viofs_serivce(test, params, guest_session)
+                # for windows guest, disable/uninstall driver to get memory leak
+                # based on driver verifier is enabled
+                win_driver_utils.memory_leak_check(vm, test, params)
+            else:
+                utils_disk.umount(fs_target, fs_dest, 'virtiofs',
+                                  session=guest_session)
+                utils_misc.safe_rmdir(fs_dest, session=guest_session)
+
+        if setup_local_nfs:
+            if vm and vm.is_alive():
+                vm.destroy()
+            params["rm_mount_dir"] = params.get("nfs_mount_dir")
+            nfs_local.cleanup()
+
+        error_context.context("Delete the user(s) on host...", test.log.info)
+        for _username in username[::-1]:
```

Comment:
> hi houqi, the user can't be deleted because the virtiofsd process is started by user, I think we should destroy the vm(meanwhile the virtiofsd will quit) before delete the users, what do you think?

Hello @xiagao ,  the line147 can kill the process virtiofsd. After that, the users will be deleted.
Is okay for you?
Thanks!

Reply:
I thought it's better to destroy vm and virtiofsd will quit accordingly which is more reasonable from produce perspective, thanks.